spyderlib/baseconfig.py

```diff
@@ -204,18 +205,102 @@ def get_image_path(name, default="not_found.png"):
 #==============================================================================
 # Translations
 #==============================================================================
+LANG_FILE = osp.join(get_conf_path(), '.langconfig')
+DEFAULT_LANGUAGE = 'en'
+
+# This needs to be updated every time a new language is added to spyder.
+LANGUAGE_CODES = {'en': u'English',
+                  'fr': u'Français',
+                  'es': u'Español',
+                  'pt_BR': u'Português (brasileiro)'
+                  }
+
+
+def get_available_translations():
+    """
+    List available translations for spyder based on the folders found in the
+    locale folder.
+    """
+    locale_path = get_module_data_path("spyderlib", relpath="locale",
+                                       attr_name='LOCALEPATH')
+    listdir = os.listdir(locale_path)
+    langs = [d for d in listdir if osp.isdir(osp.join(locale_path, d))]
+    langs = [DEFAULT_LANGUAGE] + langs
+
+    # Check that there is a language code available in case a new translation
+    # is added, to ensure LANGUAGE_CODES is updated.
+    for lang in langs:
+        if lang not in LANGUAGE_CODES:
+            error = _('Update LANGUAGE_CODES if a new translation has been '
+                      'added to Spyder')
+            raise Exception(error)
+    return langs
+
+
+def get_interface_language():

```

Comment:
Yes it might be too much, but why doing stuff by hand, if a function can handle it?

I would really like to have many more translations